site/source/tutorials/1010_containers_and_injecting_state/index.md

```diff
@@ -0,0 +1,175 @@
+
+{% include_codefile 'demo/finished/biz-e-corp/src/ApplicationContext.ts' lines:26-35 %}
+
+The `formInput` method provides the same functionality as the `_onFormInput` method in the `App` class and the `submitForm` method is analogous to `_addWorker` from the `App` class. The implementations vary slightly as the `ApplicationContext` has dedicated fields to store the state information. Also, since the `ApplicationContext` is not a widget, it cannot call `invalidate();` to schedule a re-render, the instance needs to emit an `invalidate` event that instructs associated widgets to `invalidate` themselves.
+
+Notice that the `ApplicationContext` does not contain any code to load state information. Currently its only role is only to manage the application's state provided on initialization via it's `constructor`. However as the requirements for the application become more advanced the `ApplicationContext` could make requests to fetch and modify data from a remote service or local storage mechanism.
+
+Now that we have moved state management to a dedicated module, we need a way to initialize the state and connect it to sections of our application. We will initialize state by creating and registering an `Injector`.
+
+{% section %}
+
+## Injectors
+
+{% task 'Register an `Injector` that will allow state to be injected into widgets.' %}
+
+Currently, the application's `main` module is only responsible for creating the `Projector`, which provides the bridge between the application code and the DOM.
+
+{% include_codefile 'demo/initial/biz-e-corp/src/main.ts' %}
+
+Now, we need to update `main` to construct and populate the an instance of `ApplicationContext`, create a `registry` and then define an `Injector` with the initialized `ApplicationContext`. Lastly, in order to make the `registry` available within widget tree we need to pass the `registry` as a property to the `projector`.
+
+{% instruction 'Import the `ApplicationContext` module and add this code to `main`:' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' lines:10-29 %}
+
+{% aside 'Loading data' %}
+In a real-world application, this data would probably be loaded via a call to a web service or a local data store. To learn more, take a look at the [stores tutorial](../comingsoon.html).
+{% endaside %}
+
+The state stored in the `ApplicationContext` is the same data used in the `App` module to initialize the `WorkerProperties`, but it is now decoupled into an isolated module that helps to understand and maintain the application. In general, the `main` module of an application should be concerned with initializing application wide state. Also, as we mentioned earlier, the `App` class only needed to manage the `WorkerProperties` state so that it could coordinate this through from and to it's children.
```

Comment:
"Also, as we mentioned earlier," -> "Also, as previously mentioned,"

Reply:
"coordinate this through from and to it's children" too many pronouns, let's be more explicit... also would be its children